### ISSUE TITLE
AutoroutePart on Taxonomy Term Content Type not Registering Route

### ISSUE THREAD

**Issue by davidhayden** [OrchardCMS/OrchardCore#2755]
When I add an Autoroute Part to a Content Type that serves as the Term Content Type of a Taxonomy and go to the route Orchard Core displays "Page Not Found." I assume this is a bug and this route should display the detail view of the taxonomy term. I was planning on using this route to display all the content items that are associated with the taxonomy term.

Not sure if this issue is related to https://github.com/OrchardCMS/OrchardCore/issues/2688

**Comment by sebastienros** [comment]
Known limitation, but I decided to merge still while I am working on it:

1- AutoroutePart in Terms

I worked a few days on that, and I got it working in a separate branch for now (sebros/routable). The issue is that AutoroutePart and contained items are not designed to work together. I had to introduce another concept (routable) but there is some code duplicate with autoroute, so I need to refactor it and as it will break autoroutes, I will have to provide a migration path. So this is not yet ready unfortunately.

2- using this route to display all the content items that are associated with the taxonomy term

I think this is not that easy, and we might have to find a solution that would fix that and also the previous point. The issue is that we could definitely display all the content items that are associated with a term though a Content Type could have two fields (Main Color, Other Colors) or different content types might have fields for the same taxonomy. When you display a term and you expect to see what items are related, it will then make sense to target only one single Taxonomy Field.

A solution I have in mind would be to create a new element, not a content item, like a routable thing, that would only work with a specific Taxonomy Field. We could call these "Term Pages", then select a field of a content type, and specify a route, and maybe a pattern like for autoroute, and if inherited terms are valid. This would then handle each term of the taxonomy associated with the field, and also would display the content items for this field. We would also provide alternates for these pages, and the content items inside. 

There might still be an issue with how to build the slug of each term, which would still require some of the changes that I have on the other branch.
